todo_app/templates/index.html

```diff
@@ -23,24 +67,79 @@ <h1 class="display-4">To-Do App</h1>
       </tr>
-       {% for item in items %}
-       
+      {% if items[0].showItemsToDo == 'showToDoItemsTrue' %}
+      <tr>
+        <th>Items to do</th>
+      </tr>
+      {% endif %}    
+          {% for item in items %}
+            {% if item.progress['progress'] == 'todo' %}
+              <tr>
+                <td>{{ item.idShort }}</td>
+                <td>{{ item.status }}</td>
+                <td>{{ item.card_name }}</td> 
+                <td>
+                  
+                  <FORM action="http://localhost:5000/next_list?id={{ item.card_id }}&next_idList={{ item.progress['id'] }}&todo_checkbox={{ item.showItemsToDo }}&progressing_checkbox={{ item.showProgressing }}&done_checkbox={{ item.showCompleted }}&doneitems={{ items[0].showCompletedRadio }}" method="post">
+                    <INPUT type="submit" value="{{ item.progress['name'] }}">
+                  </FORM>
+                    
+
+                </td>
+                <td>
+                  <FORM action="http://localhost:5000/delete?id={{ item.card_id }}&todo_checkbox={{ items[0].showItemsToDo }}&progressing_checkbox={{ item.showProgressing }}&done_checkbox={{ item.showCompleted }}&doneitems={{ items[0].showCompletedRadio }}" method="post">
+                    <INPUT type="submit" value="Delete">
+                  </FORM>
+                </td>
+              </tr>
+            {% endif %} 
+            {% endfor %} 
+        {% if items[0].showProgressing == 'showProgressingItemsTrue' %}    
+         <tr>
+          <th>Items that are in progress</th>
+        </tr>
+        {% endif %} 
         <tr>
+          {% for item in items %}
+          {% if item.progress['progress'] == 'inprogress' %}
           <td>{{ item.idShort }}</td>
            <td>{{ item.status }}</td>
            <td>{{ item.card_name }}</td> 
            <td>
-             <FORM action="http://localhost:5000/completed?id={{ item.card_id }}&next_idList={{ item.NextidList['id'] }}" method="post">
-                <INPUT type="submit" value="{{ item.NextidList['name'] }}">
-             </FORM>
+             
+               <FORM action="http://localhost:5000/next_list?id={{ item.card_id }}&next_idList={{ item.progress['id'] }}&todo_checkbox={{ items[0].showItemsToDo }}&progressing_checkbox={{item.showProgressing }}&done_checkbox={{item.showCompleted}}&doneitems={{ items[0].showCompletedRadio }}" method="post">
+                  <INPUT type="submit" value="{{ item.progress['name'] }}">
+                </FORM>
+               
+
            </td>
            <td>
-            <FORM action="http://localhost:5000/delete?id={{ item.card_id }}" method="post">
-               <INPUT type="submit" value="delete">
+            <FORM action="http://localhost:5000/delete?id={{ item.card_id }}&todo_checkbox={{ items[0].showItemsToDo }}&progressing_checkbox={{ item.showProgressing }}&done_checkbox={{ item.showCompleted }}&doneitems={{ items[0].showCompletedRadio }}" method="post">
```

Comment:
This shouldn't be localhost:5000 because this app should be able to run with any host. This will break in later modules, when you try hosting it on a remote server.

It can either be a relative URL, like `action="/delete?id={{ item.card_id }}etc..."` or you can use Flask's url_for: `action="{{ url_for('delete') }}?id=...etc"`